src/ptrace.h

```diff
@@ -36,4 +36,9 @@ std::string PtracePeekString(pid_t pid, unsigned long addr);
 // peek some number of bytes
 std::unique_ptr<uint8_t[]> PtracePeekBytes(pid_t pid, unsigned long addr,
                                            size_t nbytes);
+
+#ifdef __amd64__
```

Comment:
I think it's more correct to put guards into the `configure.ac` script, but I can handle that later if the rest of the code looks good.